documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+selected for bulk update), has significant cost savings and more accurately
+models the data as a single decision.
+
+#### `action`
+
+This is a slug column describing the action taken. In addition to the action
+types defined in the related IP, we will add two new options:
+
+- `reversed_deindex` - Undo a previous deindexing decision.
+- `reversed_mark_sensitive` - Undo a previous decision marking the applicable
+  records sensitive.
+
+For consistency we will also update the existing action type
+`confirmed_sensitive` to `marked_sensitive` to describe the effect of the action
+without the use of the word "confirm", which is accurate when describing single
+decisions made by moderators but potentially inappropriate for describing bulk
+actions that do not include individual review of records.
+
+## Actions
+
+The related IP describes adding new custom Django
+[admin actions](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.html#actions)
+for each of the action types described in that IP. These actions are called from
+the Report view and take action on a list of MediaReports to create a
+ModerationDecision. They do not need to be modified, except to use the
+many-to-many `media_identifiers` column (although in this workflow only a single
+record will ever be added.)
+
+We will need to add new actions for making moderation decisions directly from a
+list of Media, as well as two new actions for the `reversed_<x>` action types.
+
+_All_ actions go through an intermediate confirmation page which:
+
+- Indicates the number of records that will be affected by the action
+- Collects the required `explanation` text (all other columns, such as
+  `moderator_id` and `created_on`, are filled in automatically)
+
+### Bulk Mark Sensitive: mark the works as sensitive
+
+This action takes a queryset of `AbstractMedia` records. It creates a
+ModerationDecision with type `marked_sensitive` and adds all records from the
+queryset **which do not already have a related `SensitiveMedia` record**. It
+then creates and saves `SensitiveMedia` records for each of these.
+
+The confirmation page for this action should indicate the number of records that
+will actually be affected. If some records are already marked sensitive, that
+should be indicated on the confirmation page as well.
+
+### Bulk Deindex sensitive: entirely exclude the works from Openverse due to sensitivity
+
+This action takes a queryset of `AbstractMedia` records. It creates a
+ModerationDecision with type `deindexed_sensitive` and adds all records from the
+queryset. It then creates and saves `DeletedMedia` records for each of these,
+deleting the record from the API and deindexing it in elasticsearch. The
+confirmation page for this action should include a highly visible warning that
+once records are deindexed, they cannot be restored immediately.
```

Comment:
Should we also wait some time (10 minutes?) before deindexing, since this is such a drastic change that would require a long time to reverse.

Reply:
When I initially responded I did not see Sara's additional comment in this thread, just wanted to acknowledge that I agree with this reasoning.

Also wanted to add another mention of my [earlier comment](https://github.com/WordPress/openverse/pull/3719/files/e14e2465ba6a40bd9b1088cdbfa6f914aa2067f4#r1484704393) -- in the future I think we could implement support for mini/targeted data refreshes that would make this even quicker/easier.